### PR TITLE
Uos s3 prepare1 push

### PR DESCRIPTION
**Pull request by fyin1**

### CODE REVIEW

**Comment by NanlinXie** [comment]
Can one of the admins verify this patch?

File: hypervisor/arch/x86/guest/vlapic.c [hunk 1]
```diff
@@ -1513,9 +1513,14 @@ void
...
 	struct lapic *lapic;
+	void *apic_page;
 
 	lapic = vlapic->apic_page;
-	memset(lapic, 0, sizeof(struct lapic));
+
+	apic_page = (void *)vlapic->apic_page;
+	memset(apic_page, 0, CPU_PAGE_SIZE);
+	if (vlapic->pir_desc)
+		memset(vlapic->pir_desc, 0, sizeof(struct pir_desc));
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/vlapic.c]: This is different with what you submitted in mailinglist. This one is more old. Please update

**Comment by lijinxia** [comment]
@fyin1 Could you please rebase and fix conflicts?
Conflicting files
hypervisor/arch/x86/guest/vlapic.c

File: hypervisor/arch/x86/guest/vlapic.c (hunk 1 above)
Comment (fyin1) [reply hypervisor/arch/x86/guest/vlapic.c]: Oops. I applied the V1 when adding Acked-by. Will fix it with new PR.

Comment (fyin1) [reply hypervisor/arch/x86/guest/vlapic.c]: Applied the correct patch which match the one acked on mail list.

**Comment by lijinxia** [comment]
Verified+1

**Pull request merged**